### ISSUE TITLE
mania-stage-light is displayed under mania-stage-hint

### ISSUE THREAD

**Issue by enoslayd** [ppy/osu#9546]
**Describe the bug:** mania-stage-light is displayed under mania-stage-hint instead of above like in stable

**Screenshots or videos showing encountered issue:** 
lazer:
![osu_2020-07-13_20-41-05](https://user-images.githubusercontent.com/41866328/87306204-40954e80-c54a-11ea-995a-7a277b714cb9.png)

stable:
![screenshot068](https://user-images.githubusercontent.com/41866328/87305977-ea281000-c549-11ea-8150-c6d91f416c64.png)

**osu!lazer version:** 2020.712.0

**Logs:**
_not relevant imo but_
[runtime.log](https://github.com/ppy/osu/files/4912503/runtime.log)
[updater.log](https://github.com/ppy/osu/files/4912504/updater.log)
[database.log](https://github.com/ppy/osu/files/4912505/database.log)
[network.log](https://github.com/ppy/osu/files/4912507/network.log)

edit: wrong screenshot

**Comment by enoslayd** [comment]
Skin: [oks.zip](https://github.com/ppy/osu/files/4912515/oks.zip)

**Comment by peppy** [comment]
Doesn't the first look more correct?

**Comment by enoslayd** [comment]
Stage-light should be in front of stage-hint. In my opinion, It provides the visual "response" i need when playing.

**Comment by peppy** [comment]
I guess for now we should match stable behaviour and then offer better customisation going forward.

**Comment by enoslayd** [comment]
Yes! Thank you

**Comment by peppy** [comment]
The mania column code is... pretty hard to follow. Proxying, nested skinnables.

To fix this, lighting (currently inside `LegacyColumnBackground`) needs to be above `HitTarget` (in `ColumnHitObjectArea`). 

- Should it be moved out of the background layer? If so, should it be inside the Column, or the `HitObjectArea`? Looks like if it's in the column alone it may not be able to update its position on flip
- Should the HitTarget be moved into the background layer?

**Comment by smoogipoo** [comment]
Stable's sprite layout looks like:

```
- ContainerBelow
    - stage: mania-stage-left
    - stage: mania-stage-right
    - stage: mania-stage-light
- Container
    - stage: mania-stage-hint
    - stage: judgementLine
    - column: leftColumnLine
    - column: columnBackground
    - column: rightColumnLine
    - column: mania-keyN (if KeysUnderNotes == true)
    - column: mania-keyND (if KeysUnderNotes == true)
    - column: mania-stage-light
- ContainerNotes
- ContainerAbove
    - stage: mania-stage-bottom
    - stage: judgementText
    - stage: HD/FI masks
    - column: mania-keyN (if KeysUnderNotes == false)
    - column: mania-keyND (if KeysUnderNotes == false)
    - column: keyText
    - column: lightingL
    - column: lightingN
```

So it seems like all stage-level sprites should be moved to the background, including the `mania-stage-hint` which is the problematic sprite here.

It shouldn't be inside `Column`. It should probably be moved from `ColumnHitObjectArea` to a new `StageHitObjectArea`.

**Comment by smoogipoo** [comment]
I've started to look into this issue, and I've gone back to stable because the above didn't really make sense. There's another depth value alongside each of those. 

Taking that depth value into consideration makes it look like:
```
- ContainerBelow:
    - stage: mania-stage-light
    - stage: mania-stage-left
    - stage: mania-stage-right
- Container
    - column: columnBackground
    - column: leftColumnLine
    - column: rightColumnLine
    - stage: mania-stage-hint
    - stage: judgementLine
    - column: mania-stage-light
    - column: mania-keyN (if KeysUnderNotes == true)
    - column: mania-keyND (if KeysUnderNotes == true)
- ContainerNotes
- ContainerAbove
    - stage: HD/FI masks
    - column: mania-keyN (if KeysUnderNotes == false)
    - column: mania-keyND (if KeysUnderNotes == false)
    - column: keyText
    - stage: mania-stage-bottom
    - column: lightingL
    - column: lightingN
    - stage: judgementText
```
Things are interspersed really badly.

**Comment by enoslayd** [comment]
@smoogipoo question, if stage light is in container below and stage hint is in container,  shouldn't light be below?

**Comment by smoogipoo** [comment]
The mania-stage-light in `ContainerBelow` is hidden, it was supposed to provide a sort of "wave" effect but it wasn't ever fully implemented iirc. Here's how it'd look:
![image](https://user-images.githubusercontent.com/1329837/91048793-24132880-e657-11ea-8108-eafc56a606f9.png)
So for the purposes of this it can be disregarded.

**Comment by enoslayd** [comment]
Oh, there was two stage light thats why im confused
